### PR TITLE
Suppress existing WAL warning when archive-mode-check is disabled.

### PR DESCRIPTION
**Pull request by jreidthompson**
Do not emit duplicate WAL warning from archive-push command when archive-mode-check is disabled.

See card
https://github.com/pgbackrest/pgbackrest/projects/2#card-66266515

### CODE REVIEW

File: src/command/archive/push/file.h [hunk 1]
```diff
@@ -32,6 +32,7 @@ typedef struct ArchivePushFileResult
 // Copy a file from the source to the archive
 ArchivePushFileResult archivePushFile(
     const String *walSource, bool headerCheck, unsigned int pgVersion, uint64_t pgSystemId, const String *archiveFile,
-    CompressType compressType, int compressLevel, const List *const repoList, const StringList *const priorErrorList);
+    CompressType compressType, int compressLevel, const List *const repoList, const StringList *const priorErrorList,
+    bool modeCheck);
```
Comment (dwsteele) [inline src/command/archive/push/file.h:37]: Pointers should not be marked as const in the header.

**Review by dwsteele — CHANGES_REQUESTED** [review]
Looks pretty good, see comments.

File: src/command/archive/push/file.c [hunk 2]
```diff
@@ -94,8 +94,9 @@ archivePushFileIo(ArchivePushFileIoType type, IoWrite *write, const Buffer *buff
 /**********************************************************************************************************************************/
 ArchivePushFileResult
 archivePushFile(
-    const String *walSource, bool headerCheck, unsigned int pgVersion, uint64_t pgSystemId, const String *archiveFile,
-    CompressType compressType, int compressLevel, const List *const repoList, const StringList *const priorErrorList)
+    const String *walSource, const bool headerCheck, const unsigned int pgVersion, const uint64_t pgSystemId,
+    const String *archiveFile, CompressType compressType, const int compressLevel, const List *const repoList,
+    const StringList *const priorErrorList, const bool modeCheck)
 {
```
[the remark this answers is not in the document]
Comment (jreidthompson) [reply src/command/archive/push/file.c:100]: 34b68add moves modeCheck to follow headerCheck

File: src/command/archive/push/file.h (hunk 1 above)
Comment (jreidthompson) [reply src/command/archive/push/file.h:37]: cd7c88a0 unmarks pointers as const in header

**Review by dwsteele — APPROVED** [review]
Add a few missing const keywords in 5392fad67214fced.

**Pull request merged**